client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -3,156 +3,74 @@
-			return true;
-		}
-		return false;
-	};
+	const handleNext = () => setActiveStep( activeStep + 1 );
+
+	const steps = [
+		{
+			value: siteType,
+			label: isFilledFormValue( siteType )
+				? NO__( 'I want to create a website ' )
+				: NO__( 'Setting up a website ' ),
+			displayValue: isFilledFormValue( siteType ) ? siteTypeOptions[ siteType ] : '',
+			StepInput: SiteTypeSelect,
+		},
+		{
+			value: siteVertical,
+			label: NO__( 'My site is about' ),
+			displayValue: isFilledFormValue( siteVertical )
+				? siteVertical.label
+				: NO__( 'enter a topic' ),
+			StepInput: VerticalSelect,
+		},
+		{
+			value: siteTitle,
+			label: NO__( "It's called" ),
+			displayValue: siteTitle.length ? siteTitle : NO__( 'enter a title' ),
+			StepInput: SiteTitle,
+		},
+	];
 
 	return (
 		<div className="onboarding-block__acquire-intent">
 			<div className="onboarding-block__questions">
 				<h2 className="onboarding-block__questions-heading">
-					{ isEmpty( siteType ) && NO__( "Let's set up your website – it takes only a moment." ) }
+					{ ! isFilledFormValue( siteType ) &&
+						NO__( "Let's set up your website – it takes only a moment." ) }
 				</h2>
-
-				<div
-					className={ classNames( {
-						'onboarding-block__question': true,
-						selected: isSelected( 'siteType' ),
-					} ) }
-				>
-					<span>
-						{ isEmpty( siteType )
-							? NO__( 'I want to create a website ' )
-							: NO__( 'Setting up a website ' ) }
-					</span>
-					{ isSelected( 'siteType' ) ? (
-						<ul className="onboarding-block__multi-question">
-							{ map( siteTypeOptions, ( label, value ) => (
-								<li key={ value } className={ value === siteType ? 'selected' : '' }>
-									<label>
-										<input
-											name="onboarding_site_type"
-											onChange={ selectSiteType }
-											type="radio"
-											value={ value }
-										/>
-										<span className="onboarding-block__multi-question-choice">{ label }</span>
-									</label>
-								</li>
-							) ) }
-						</ul>
-					) : (
-						<div className="onboarding-block__multi-question">
-							<button className="onboarding-block__question-answered" onClick={ openSiteType }>
-								{ siteTypeOptions[ siteType ] }
-							</button>
-							<span>.</span>
-						</div>
-					) }
-				</div>
-
-				{ ( isFilledFormValue( siteType ) || ! isEmpty( siteVertical ) ) && (
-					<div
-						className={ classNames( {
-							'onboarding-block__question': true,
-							selected: isSelected( 'siteVertical' ),
-						} ) }
-					>
-						<span>{ NO__( 'My site is about' ) }</span>
-						<div className="onboarding-block__multi-question">
-							<VerticalSelect
-								onClick={ selectSiteVertical }
-								forceHideSuggestions={ ! isSelected( 'siteVertical' ) }
-								inputClass="onboarding-block__question-input"
+				{ steps.map(
+					( step, index ) =>
+						( index < 1 || isFilledFormValue( steps[ index - 1 ].value ) ) && (
+							<Question
+								key={ index }
```

Comment:
https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318 :grimacing:

Reply:
In this case, our steps are statically defined and even if it may seem we are filtering them, they are always displayed in the same order so I'm pretty sure React will not get crazy.

This kind of exception is also mentioned in the article above:
> - the list and items are static–they are not computed and do not change;
> - the items in the list have no ids;
> - the list is never reordered or filtered.
>
> When all of them are met, you may safely use the index as a key.